### PR TITLE
Fix SVG validation by supporting XML prolog and leading comments

### LINKED ISSUE #7460: PIXI.Loader cannot load SVG with XML prolog
### Current Behavior
Any SVG with an XML prolog (such as default exports from Boxy SVG) cannot be used with PIXI.Loader. The SVG data is preloaded via XHR and fails the SVGResource.test() funciton. 

SVGs with XML prologs can still be used as normal with PIXI.Texture.from() because in this case it is the SVG URL and not the file data that is passed to SVGResource.test().

### Possible Solution
Update SVGResource to accept SVGs with XML prologs. The test function will require an additional regex test and the load function will also require updates.

### Steps to Reproduce
Load any SVG file with an XML prolog (see below) included at the start of the file using the PIXI.Loader add function.

image

Tested on Pixi.js 6.0.2

### PR DESCRIPTION
**Pull request by SukantPal**
Fixes #7460, #7477

##### Description of change

`SVGResource.test` strips out the prolog and leading comments before checking for the first tag being `<svg>`.

##### Pre-Merge Checklist

- [x] Tests and/or benchmarks are included
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
What you're doing seems like a lot of work. Why not replace this:

```js
source.indexOf('<svg') === 0
```

with a more robust regexp:

```js
/^(<\?xml[^?]+\?>)?\s*(<!--[^(-->)]*-->)?\s*\<svg/gm.test(source)
```

This should match the following:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<!-- Hello -->
<svg>
  Hello world
</svg>
```

```xml
<?xml version="1.0" encoding="utf-8" ?>
<svg>
  Hello world
</svg>
```

```xml
<svg>
  Hello world
</svg>
```

**Comment by SukantPal** [comment]
Yep, looks like I wasn't thinking right last night 😅. Simplified it now.

**Comment by bigtimebuddy** [comment]
Why String.match instead of RegExp.test? Test returns Boolean, match returns an array.

File: packages/core/src/textures/resources/SVGResource.ts [hunk 1]
```diff
@@ -245,9 +245,16 @@ export class SVGResource extends BaseImageResource
             // source is SVG data-uri
             || (typeof source === 'string' && (/^data:image\/svg\+xml(;(charset=utf8|utf8))?;base64/).test(source))
             // source is SVG inline
-            || (typeof source === 'string' && source.indexOf('<svg') === 0);
+            || (typeof source === 'string' && !!source.match(SVGResource.SVG_XML));
```
Comment (SukantPal) [inline packages/core/src/textures/resources/SVGResource.ts:248]:
```suggestion
            || (typeof source === 'string' && SVGResource.SVG_XML.test(source);
```

File: packages/core/src/textures/resources/SVGResource.ts [hunk 2]
```diff
@@ -245,9 +245,16 @@ export class SVGResource extends BaseImageResource
             // source is SVG data-uri
             || (typeof source === 'string' && (/^data:image\/svg\+xml(;(charset=utf8|utf8))?;base64/).test(source))
             // source is SVG inline
-            || (typeof source === 'string' && source.indexOf('<svg') === 0);
+            || (typeof source === 'string' && SVGResource.SVG_XML.test(source);
```
Comment (SukantPal) [inline packages/core/src/textures/resources/SVGResource.ts:248]:
```suggestion
            || (typeof source === 'string' && SVGResource.SVG_XML.test(source));
```

File: packages/core/src/textures/resources/SVGResource.ts [hunk 3]
```diff
@@ -245,9 +245,16 @@ export class SVGResource extends BaseImageResource
...
-            || (typeof source === 'string' && source.indexOf('<svg') === 0);
+            || (typeof source === 'string' && SVGResource.SVG_XML.test(source));
     }
 
+    /**
+     * RegExp for SVG XML document.
+     *
+     * @example &lt;?xml version="1.0" encoding="utf-8" ?&gt;&lt;!-- image/svg --&gt;&lt;svg
+     */
+    static SVG_XML = /^(<\?xml[^?]+\?>)?\s*(<!--[^(-->)]*-->)?\s*\<svg/gm;
```
Comment (SukantPal) [inline packages/core/src/textures/resources/SVGResource.ts:256]: Can't use global flag for shared RegExp @bigtimebuddy :(

```suggestion
    static SVG_XML = /^(<\?xml[^?]+\?>)?\s*(<!--[^(-->)]*-->)?\s*\<svg/m;
```

**Review by Zyie — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
Thanks @SukantPal!

**Pull request merged**